cmake/toolchain/zephyr/Kconfig

```diff
@@ -0,0 +1,10 @@
+# Zephyr Toolchain
```

Comment:
Should this be here? Or should this just continue to use the bits provided in the SDK itself?

Reply:
The intent is to be able to put things here that either don't depend on the SDK version, or we don't want to have to update and re-spin the SDK for.

So for example when you introduce `TOOLCHAIN_SUPPORTS_NEWLIB`, that can just go in here w/o having to update the SDK Kconfig.

Reply:
all good, as long as the sdk team is aligned.

Reply:
This seems to have a built-in risk of getting back to the mess we used to have, but was cleaned up here: https://github.com/zephyrproject-rtos/zephyr/commit/e747fe73cdd67459d268e7ea2770a513310373ce

> The intent is to be able to put things here that either don't depend on the SDK version

How do you know ?
If there is a setting here describing a feature of the SDK, why can't that feature be removed in a future version ?
Or in case the Zephyr-SDK is split into multiple packages, that the feature is only available if a specific sub-package is installed.

> or we don't want to have to update and re-spin the SDK for.

has this been a problem so far ?

Now you introduce Kconfig files in two places that describes the toolchain.
In Zephyr tree, and and in the SDK.
At some point someone will end-up placing a setting in the wrong file (most likely in the Zephyr repo, when it should have been in the Sdk repo), and that PR just gets approved because it looks ok, and things are working with the current SDK.

Not to mention, there are now to env settings for kconfig that controls Kconfig sourcing:
- `ZEPHYR_TOOLCHAIN_KCONFIG_DIR=${ZEPHYR_TOOLCHAIN_KCONFIG_DIR}`
- `TOOLCHAIN_KCONFIG_DIR=${TOOLCHAIN_KCONFIG_DIR}`

bound to result in confusion.

I'm not convinced, and I fear there's a high risk of ending up in a situation we had we Zephyr SDK and versions years back.

Reply:
I think we should have something like this since we shouldn't require a user to have to download a new toolchain just because of a Kconfig update.  For example when we introduce TOOLCHAIN_HAS_PICOLIBC.

I get that there might be some confusion on where to put what Kconfig's but, I'd argue that we don't do that very often, we have people to review those changes to tell a developer if they are placing it in the wrong place.

Reply:
How can we introduce a setting like `TOOLCHAIN_HAS_PICOLIBC` into an already released toolchain?  My point is we need Kconfig in both Zephyr and the SDK to handle cases in which the Zephyr code base wants to describe a new Kconfig symbol for something that exists in an already released toolchain, and spinning a new toolchain version just to update the Kconfig doesn't make sense to me.

Reply:
> How can we introduce a setting like TOOLCHAIN_HAS_PICOLIBC into an already released toolchain?

will such a feature be introduced into Zephyr SDK without it being actively used ?

>  My point is we need Kconfig in both Zephyr and the SDK to handle cases in which the Zephyr code base wants to describe a new Kconfig symbol for something that exists in an already released toolchain

and that's why there should be a common toolchain file which simply defines the settings and type of toolchain features, and then dedicated toolchains can describe the default without specifying the type, see other comment.

We don't need an extra setting and Kconfig file for the Zephyr SDK for this.
If for some reason there really is a high demand for a temp solution until a new SDK spin is ready, or backport reasons where older SDK may also support a feature we start using, then we may have a `default y if TOOLCHAIN_ZEPHYR_0_16` or similar, and then remove this special condition handling at earliest possibility.

I still don't see a justification for having an extra Zephyr SDK Kconfig tree and setting which must be passed to konfiglib invocation, considering the risk of misuse / mistakes that may occur.